app/lib/actions/katello/event_queue/monitor.rb

```diff
@@ -0,0 +1,60 @@
+module Actions
+  module Katello
+    module EventQueue
+      class Monitor < Actions::EntryAction
+        include ::Dynflow::Action::Singleton
+
+        Handle = Algebrick.type do
+          fields! event_type: String, object_id: Integer
+        end
+
+        Poll = Algebrick.atom
+
```

Comment:
Unless stated otherwise, actions use the `default` queue. The `default` sidekiq queue is already used for almost everything, is it ok that event processing might be delayed?

Reply:
It should be OK in combination with your prior point: instead of using a Poll -> Handle ping-pong the Poll will fully drain the queue in a loop which can be broken early with `if world.terminating?` to make sure shutdown isn't blocked.